### PR TITLE
[Innawood] Chemistry, electricity, and more!

### PR DESCRIPTION
**Pull request by Light-Wave**
#### Summary
Mods "[Innawood] Chemistry, electricity, and more!"

#### Purpose of change
The main purpose of this pull request is to expand the end game of the Innawood mod, with focus on chemistry and electricity.
This PR also includes a wild mix of minor features.

#### Describe the solution
Everything is contained to the Innawood mod.
- Book hating, illiterate, and churl has had their point value nerfed to reflect the lack of books.
- You can now make makeshift transformers, which can be used instead of power_supply, e_scrap, and amplifier in many recipes. Perhaps most notable, this makes soldering irons and makeshift arc welders craftable.
- Can make light bulbs from scratch, allowing you to make light producing tools and vehicle parts
- A new item, a car battery with a shell made of clay, is now craftable. Can be installed in vehicles.
- Trapdoor spiders have been nerfed. The map extra has had the spawns cut in half (15, down from 30), and bones has been added to the surrounding ground, warning players that something is not quite right. Furthermore, the trap spawning them can be noticed with the same difficulty as buried landmines, and when triggered they will spawn young trapdoor spiders, instead of adult ones.
- You can now get platinum as a byproduct from smelting copper.
- Stills can now be crafted, and you can use copper pots in their recipe.
- chem_ethanol, methed_alcohol, and biodiesel can be crafted.
- Rubber, rubber cement, and their components can be crafted
- Willowbark tea is autolearn
- bleach and oxy_powder is autolearn
- formic_acid from stringers can be made with a metal funnel, instead of requiring an uncraftable funnel.
- glass shards can be made with a pot or a makeshift pot in place of sheet metal.
- many electronic recipies are autolearned, and many of them can be made with makeshift transformers instead of uncraftable items.
- Added mechanics practice using a frame and a wheel that can take you from level 0 to level 3.
- Pipe cleaners are autolearned, finally allowing you to clean your musket, instead of having to throw it away when it gets dirty.
- separation funnel, fractional distillation apparatus, and thermometer can be crafted now.
- bronze spear and sword are autolearned (called dory and xiphos) 
- makes sausages craftable with leather and metal funnel. Solves #56356

#### Describe alternatives you've considered
The light bulb could do with a more realistic recipe, but I wanted electronic light sources available.
I considered electronic scrap and circuit boards being available from portal storm events instead.
I'm not sure if I nerfed trapdoor spiders too hard. Might have.
The thermometer recipe is a bit strange, as it uses clothing dye requirements as part of the recipe. Perhaps an alternative list of requirements should be made instead.

#### Testing
Loaded up the mod and checked that everything new was craftable. Crafted several of the things added here.

### CODE REVIEW

File: data/mods/innawood/items/electronics.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    "price": 170,
+    "price_postapoc": 15,
+    "material": [ "iron", "copper" ],
+    "weight": "113 g",
+    "volume": "1 L",
+    "bashing": 5,
+    "to_hit": -1
+  },
+  {
+    "type": "recipe",
```
Comment (Sathra225) [inline data/mods/innawood/items/electronics.json:20]: Is the lightbulb recipe meant to be here?

**Comment by Sathra225** [comment]
That reminds me, the bronze weapons should probably have Grinding added to their recipes.

File: data/mods/innawood/items/electronics.json (hunk 1 above)
Comment (Light-Wave) [reply data/mods/innawood/items/electronics.json]: Good catch, moved.

**Comment by Light-Wave** [comment]
> That reminds me, the bronze weapons should probably have Grinding added to their recipes.

I'll just go ahead and throw in { "id": "GRIND", "level": 1 } on all sharp bronze objects.

**Comment by hostergaard** [comment]
There seems to be no way to achieve ethanol.

Ethanol requires a thermometer, which in turns requires ethanol to make.

Ethanol can be made as a by product of methanol. But that methanol recipe requires moonshine down the line, which can be made either with pumpkins or corn, neither which seems to be able to be found in the wild. 

So, maybe corn or pumpkin forageble or find an alternate path to ethanol without a thermometer, or a thermometer without ethanol?

**Comment by Zireael07** [comment]
> So, maybe corn or pumpkin forageble

This please

**Comment by NetSysFire** [comment]
I am pretty sure you can find those in anthills or maybe even caves.

**Comment by Light-Wave** [comment]
> I am pretty sure you can find those in anthills or maybe even caves.

That is technically true, but I did not intend for the player to need to loot anthills to make a thermometer. I will have to look into other methods of making ethanol.

**Comment by Sathra225** [comment]
Technically, you can make moonshine from fruit. Its just more likely to be toxic due to more methanol apparently. Applejack is a kind of moonshine, concentrated by freezing.
Or just allow the higher-proof booze to be used as well for thermometers. One of the first recorded thermometers used _brandy_ as the working liquid.
Third option is adding another, longer ethanol recipe for low-heat distillation and re-distillation. Basically not using a thermometer to set the temp at the right temperature (78C), but at a lower one for slow distillation (50C).

**Comment by Light-Wave** [comment]
> Not seeing Platinum as a by-product of Copper smelting so far in the recipes (which is needed for Sulfuric Acid > Formic Acid > Plastic Chunk) in Experimental [1d66199](https://github.com/CleverRaven/Cataclysm-DDA/commit/1d66199bf5869380c0c86ae7e779978544be187a).
> 
> Should I create an Issue ticket for this separately?

It's in data/mods/innawood/mining/recipes/smelting.json line 34
`"byproducts": [ [ "platinum_small" ] ],`

**Comment by evankimori** [comment]
> > Not seeing Platinum as a by-product of Copper smelting so far in the recipes (which is needed for Sulfuric Acid > Formic Acid > Plastic Chunk) in Experimental [1d66199](https://github.com/CleverRaven/Cataclysm-DDA/commit/1d66199bf5869380c0c86ae7e779978544be187a).
> > Should I create an Issue ticket for this separately?
> 
> It's in data/mods/innawood/mining/recipes/smelting.json line 34 `"byproducts": [ [ "platinum_small" ] ],`

Oof. Figured out the issue. Was testing on a build with MiningMod & MOre Survivor that alters the Innawoods Copper recipe and removes a LOT of the byproduct listings. Not sure if that's something you'd wanna bring up with that mod author as an issue. 

My bad as I forgot I was not testing on a clean Innawoods build RE: some additional content mods that would've caused conflicts. I can add some screenshots if needed of the issue but clearly this was an oversight on my end. Apologies for the inconvenience. Disregard if unnecessary.

**Pull request merged**